modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -411,23 +411,13 @@ def where(self, cond, other, **kwargs):
             cond, type(self)
         ), "Must have the same QueryCompiler subclass to perform this operation"
         if isinstance(other, type(self)):
-            # Note: Currently we are doing this with two maps across the entire
-            # data. This can be done with a single map, but it will take a
-            # modification in the `BlockPartition` class.
-            # If this were in one pass it would be ~2x faster.
-            # TODO (devin-petersohn) rewrite this to take one pass.
-            def where_builder_first_pass(cond, other, **kwargs):
-                return cond.where(cond, other, **kwargs)
-
-            first_pass = cond._modin_frame.binary_op(
-                where_builder_first_pass, other._modin_frame, join_type="left"
-            )
-
-            def where_builder_second_pass(df, new_other, **kwargs):
-                return df.where(new_other.eq(True), new_other, **kwargs)
-
-            new_modin_frame = self._modin_frame.binary_op(
-                where_builder_second_pass, first_pass, join_type="left"
+            new_modin_frame = self._modin_frame.n_ary_op(
+                lambda df, cond, other: df.where(cond, other, **kwargs),
+                [
+                    cond._modin_frame,
+                    other._modin_frame,
+                ],
+                join_type=None,
```

Comment:
What does join_type=None mean?

Reply:
It means we always use the left labels to align dataframes:

https://github.com/modin-project/modin/blob/bd326f1c4175102489f08d271a53cf374bd9125e/modin/core/dataframe/pandas/dataframe/dataframe.py#L1312-L1314

This is different from a `left` join because it uses the left index and columns exactly, whereas `left` join can change the order even if the binary operands have the same labels in different orders.

e.g. 

```python
import pandas as pd

df1 = pd.DataFrame([[2, 1]], columns=[1, 0])
df2 = pd.DataFrame([[2, 1]], columns=[0, 1])
# does not use df1 columns:
print((df1 + df2).columns)
# Uses df1 columns:
print(df1.where(df1 == 1, df2))
```

I added a comment explaining why we use `None`.